macros/math/PGnumericalmacros.pl

```diff
@@ -64,48 +63,53 @@ sub plot_list {
 		$yref = \@y_vals;
 	}
 
-	my $fun = sub {
+	return sub {
 		my $x = shift;
-		my $y;
-		my ($x0, $x1, $y0, $y1);
+		my ($y, $x0, $x1, $y0, $y1);
 		my @x_values = @$xref;
 		my @y_values = @$yref;
-		while ((@x_values and $x > $x_values[0])
-			or (@x_values > 0 and $x >= $x_values[0]))
-		{
+		while ((@x_values and $x > $x_values[0]) || (@x_values > 0 and $x >= $x_values[0])) {
 			$x0 = shift(@x_values);
 			$y0 = shift(@y_values);
 		}
-		# now that we have the left hand of the input
-		#check first that x isn't out of range to the left or right
+		# Now that we have the left hand of the input, check first that x isn't out of range to the left or right
 		if (@x_values && defined($x0)) {
 			$x1 = shift(@x_values);
 			$y1 = shift(@y_values);
 			$y  = $y0 + ($y1 - $y0) * ($x - $x0) / ($x1 - $x0);
 		}
-		$y;
+		return $y;
 	};
-	$fun;
 }
 
-=head2 Horner polynomial/ Newton polynomial
+=head3 Horner polynomial/ Newton polynomial
 
 Usage:
 
-    $fn = horner([x0,x1,x2],[q0,q1,q2]);
+    $fn = horner([x0,x1,x2, ...],[q0,q1,q2, ...]);
 
 Produces the newton polynomial
 
-    &$fn(x) = q0 + q1*(x-x0) +q2*(x-x1)*(x-x0);
+    &$fn(x) = q0 + q1*(x-x0) +q2*(x-x1)*(x-x0) + ...;
+
+Generates a subroutine which evaluates a polynomial passing through the points
+C<(x0,q0), (x1,q1), (x2, q2)>, ... using Horner's method.
 
-Generates a subroutine which evaluates a polynomial passing through the points C<(x0,q0), (x1,q1),
-... > using Horner's method.
+The array refs for C<x> and C<q> can be any length but must be the same length.
+
+=head4 Example
```

Comment:
This shouldn't be a header at all.  Change this to
```suggestion
For example,
```
Do the same with the other `=head4 Example` on line 570.  Those two headers are out of place in the sidebar.  Examples shouldn't be headers in any case.